### ISSUE TITLE
Properties of <Chart/> not updating when changed

### ISSUE THREAD

**Issue by mindjuice** [whawker/react-jsx-highcharts#45]
Hi...first, let me say thanks for this project.  I was considering doing something like this for Highcharts, because....well...declarative graphs FTW!  Now I can just use this and help make it better.

I'm running into an issue here where I want to be able to change the`backgroundColor` of the chart (among other things like margins, borders, etc.)  The problem is that if I change a prop of the `<Chart>` component, for example, the `backgroundColor`, even though the `render()` function of my component is called with the right color value, the background color of the chart doesn't change.  This happens for margins too and I expect other props.

In both cases, the initial value of the prop is respected, so the margins and background color are correct, but later changes are not rendered.

I was able to replicate this in the `AddSeries` demo by changing one of the buttons to change the backgroundColor in the state.  Here is the updated code:

```
import React, { Component } from 'react'
import Highcharts from 'highcharts'
import { Chart, HighchartsChart, withHighcharts, XAxis, YAxis, Title, Legend, LineSeries } from 'react-jsx-highcharts'
import ExampleCode from '../utils/ExampleCode'
import code from './exampleCode'
import { createRandomData } from '../utils/data-helpers'

class App extends Component {
  constructor(props) {
    super(props)
    this.createRandomSeries = this.createRandomSeries.bind(this)
    this.handleAddSeries = this.handleAddSeries.bind(this)
    this.handleRemoveSeries = this.handleRemoveSeries.bind(this)
    this.renderSeries = this.renderSeries.bind(this)

    const now = Date.now()
    this.state = {
      now,
      series: [
        {
          name: 'Profit',
          data: createRandomData(now, 1e8),
        },
      ],
      seriesCounter: 1,
      color: '#ff0000',
    }
  }

  createRandomSeries(index) {
    return {
      name: `Series${index}`,
      data: createRandomData(this.state.now, 1e8),
    }
  }

  handleAddSeries = e => {
    e.preventDefault()
    this.setState({
      color: this.state.color === '#ff0000' ? '#0000ff' : '#ff0000',
    })
  }

  handleRemoveSeries(e) {
    e.preventDefault()
    const { series } = this.state
    const randomIndex = Math.floor(Math.random() * series.length)
    series.splice(randomIndex, 1)

    this.setState({
      series,
    })
  }

  renderSeries({ name, data }) {
    return <LineSeries id={name} name={name} key={name} data={data} />
  }

  render() {
    return (
      <div className="app">
        <HighchartsChart>
          <Chart backgroundColor={this.state.color} />
          <Title>Dynamically add/remove series</Title>

          <Legend align="left">
            <Legend.Title>Legend</Legend.Title>
          </Legend>

          <XAxis type="datetime">
            <XAxis.Title>Time</XAxis.Title>
          </XAxis>

          <YAxis id="price">
            <YAxis.Title>Price</YAxis.Title>
            {this.state.series.map(this.renderSeries)}
          </YAxis>
        </HighchartsChart>

        <div className="btn-toolbar" role="toolbar">
          <button className="btn btn-primary" onClick={this.handleAddSeries}>
            Toggle Color
          </button>
          <button className="btn btn-danger" onClick={this.handleRemoveSeries}>
            Remove line series
          </button>
        </div>

        <ExampleCode name="AddSeries">{code}</ExampleCode>
      </div>
    )
  }
}

export default withHighcharts(App, Highcharts)
```

This seems like something that should work, so I'm wondering if I am missing something or if there is a bug.

Thanks!

**Comment by whawker** [comment]
You're right, there is a bug. I haven't implemented `componentDidUpdate` on the [Chart](https://github.com/whawker/react-jsx-highcharts/blob/master/packages/react-jsx-highcharts/src/components/Chart/Chart.js) component.

Sorry for the late reply, been on holiday for the last two weeks. I'll get this fixed asap.

**Comment by mindjuice** [comment]
Thanks for the quick fix!  I had worked around it for `Chart` for the moment by always re-rendering it, but I noticed that something similar is also happening for `LineSeries`, which seems more unexpected, since it does implement`componentDidUpdate()`.

I set the `color` prop on it and when the prop changes, the line doesn't update.  Not sure if it is related to the fix above somehow.  I can see the new color value in the React dev tools inspector in Chrome, so I know it's making it down to the component, but it doesn't re-render the line in the new color.

Side topic: How are you debugging the library when using it in an app?  I tried building and npm linking it in my app, but I wasn't able to open any of the `react-jsx-highcharts` source files in Chrome when debugging.

**Comment by whawker** [comment]
I've just published `2.0.2` with the above fix, although I'm not seeing the same issue with the `LineSeries` color you mention. See my example [here](https://whawker.github.io/react-jsx-highcharts/examples/UpdateChart/) (using [this code](https://github.com/whawker/react-jsx-highcharts/blob/master/examples/UpdateChart/App.js)).

I usually render the chart with a `<Debug />` component, which console logs the underlying chart. Then I compare the React dev tools inspector state, with the underlying chart's state.

Not the most efficient, but does the job for me. 

```jsx
<HighchartsChart>
  <Chart />
  // etc...
  <Debug />
</HighchartsChart>
```

**Comment by mindjuice** [comment]
Confirmed that this worked for the chart properties in my project.  Thanks for getting it fixed (and released) so quickly!

The other issue I had where `LineSeries` components were not updating was because I hadn't added `id` props to them.  The props are updating when the values change now.
